pkg/nodediscovery/nodediscovery.go

```diff
@@ -174,7 +174,8 @@ func (n *NodeDiscovery) StartDiscovery() {
 	defer n.localNodeLock.Unlock()
 
 	n.fillLocalNode()
-
+	n.Manager.NodeUpdated(n.localNode)
+	// Ensure local nodeUpdates are completed before we add local node to cluster.
```

Comment:
Optional docs improvement
```suggestion
	// Ensure local nodeUpdates are completed before adding the local node to the cluster.
```